PREP_SYSTEM_FOR_DIETPI.sh

```diff
@@ -513,9 +504,17 @@ _EOF_
 		aPACKAGES_REQUIRED_INSTALL+=('intel-microcode')
 		aPACKAGES_REQUIRED_INSTALL+=('amd64-microcode')
 		aPACKAGES_REQUIRED_INSTALL+=('firmware-linux-nonfree')
-		aPACKAGES_REQUIRED_INSTALL+=('firmware-misc-nonfree')
-		aPACKAGES_REQUIRED_INSTALL+=('grub-efi-amd64')
+		#aPACKAGES_REQUIRED_INSTALL+=('firmware-misc-nonfree')
 		#aPACKAGES_REQUIRED_INSTALL+=('dmidecode')
+		if (( $(fdisk -l $1 | grep "Disklabel type:" | awk '{ print $3 }') == 'dos' )); then
+
+			aPACKAGES_REQUIRED_INSTALL+=('grub2')
+
+		else
+
+			aPACKAGES_REQUIRED_INSTALL+=('grub-efi-amd64')
+
+		fi
```

Comment:
@MichaIng 
Smart detection of BIOS/EFI 👍

Reply:
I hope the guys here was right, that != dos always means GPT = efi 😆: https://unix.stackexchange.com/a/328734

Reply:
@Fourdee 
Just found another solution, that looks a bid more secure to me: https://askubuntu.com/a/162896
I guess this is something we need to test, but generally both should work.

Reply:
Yep, another one, this directory will always exist on EFI system (AFAIK):
```
if [ -d /boot/efi ]; then
```

Reply:
@Fourdee 
I just tested to install fresh Debian Jessie on VirtualBox with UEFI and install DietPi by new script. Somehow it still installed grub2. I tested the condition manually and Disklabel type: gpt is definitely there. Had to install grub-efi-amd64 by hand.

After reboot something went wrong with the partitions, maybe as I needed separate efi partition and we don't expect this on VM yet. Or something else with efi bootloader went wrong. Will try it tomorrow again.